examples/loader-google-bigquery/src/index.md

```diff
@@ -0,0 +1,168 @@
+
+```js run=false
+import "dotenv/config";
+import {BigQuery} from "@google-cloud/bigquery";
+
+const {BQ_PROJECT_ID, BQ_CLIENT_EMAIL, BQ_PRIVATE_KEY} = process.env;
+
+if (!BQ_PROJECT_ID) throw new Error("missing BQ_PROJECT_ID");
+if (!BQ_CLIENT_EMAIL) throw new Error("missing BQ_CLIENT_EMAIL");
+if (!BQ_PRIVATE_KEY) throw new Error("missing BQ_PRIVATE_KEY");
+
+const bigQueryClient = new BigQuery({
+  projectId: BQ_PROJECT_ID,
+  credentials: {
+    client_email: BQ_CLIENT_EMAIL,
+    private_key: BQ_PRIVATE_KEY.replace(/\\n/g, '\n')
+  }
+});
+
+export async function runQuery(query) {
+  const [rows] = await bigQueryClient.query({ query });
+  return rows;
+}
+```
+
+<div class="note">
+
+To run this data loader, you’ll need to install `@google-cloud/bigquery`, `d3-dsv`, `d3-time-format`, and `dotenv` using your preferred package manager such as npm or Yarn.
+
+</div>
+
+For the data loader to authenticate with the Google Bigquery API, you will need to set several environment variables containing secret credentials. If you use GitHub, you can use [secrets in GitHub Actions](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions) to set environment variables; other platforms provide similar functionality for continuous deployment. For local development, we use the `dotenv` package, which allows environment variables to be defined in a `.env` file which lives in the project root and looks like this:
+
+```
+BQ_PROJECT_ID="123456789-abc"
+BQ_CLIENT_EMAIL="xxx@yyy.iam.gserviceaccount.com"
+BQ_PRIVATE_KEY="-----BEGIN PRIVATE KEY-----\nxxxxxxxxxx-----END PRIVATE KEY-----\n"
+```
+
+<div class="warning">
+
+The `.env` file should not be committed to your source code repository; keep your credentials secret.
+
+</div>
+
+<div class="note">
+
+See the [Google Bigquery API Quickstart](https://cloud.google.com/bigquery/docs/authentication) for how to create the service account needed to access the Google BigQuery API after it has been enable in the GCP console. The following Google codelab might be helpful walk-through before getting started: https://codelabs.developers.google.com/codelabs/cloud-bigquery-nodejs#0
+
+</div>
+
+The above data loader lives in `data/covidstats_it.csv.js`, so we can load the data as `data/covidstats_it.csv`. The `FileAttachment.csv` method parses the file and returns a promise to an array of objects, which gets rendered through/into `covid_2020.md`. To create the csv file, please run `node src/data/covidstats_it.csv.js` from the right directory.
+
+```js echo
+const covidStats = FileAttachment("covidstats_it.csv").csv({typed: true});
+```
+
+The `covidStats` table has two columns: `formatted_date` and `confirmed_cases`. We can display the table using `Inputs.table`.
+
+```js echo
+Inputs.table(covidStats)
+```
+
+Lastly, we can pass the table to `Plot.plot` to make a simple line chart.
+
+```js echo
+display(
+  Plot.plot({
```

Comment:
That's a bit too much code for a simple line chart, we can trim it down quite a bit (I can help with that).

Reply:
Yes, can you trim please 🙏